packages/batch-submitter/src/batch-submitter/state-batch-submitter.ts

```diff
@@ -0,0 +1,96 @@
+/* External Imports */
+import { TransactionReceipt } from '@ethersproject/abstract-provider'
+import { getContractFactory } from '@eth-optimism/contracts'
+
+/* Internal Imports */
+import { L2Block, Bytes32 } from '..'
+import { RollupInfo, BatchSubmitter } from '.'
+
+export class StateBatchSubmitter extends BatchSubmitter {
```

Comment:
TODO: Change this so that we calculate `start = scc.totalElements()` and `end = ctc.totalElements()` ! Not based on the length of the L2 chain -- that is only used in the batch submitter